### PR TITLE
Implement middleware for authentication and abstracted db calls with DAO 

### PR DESCRIPTION
**Pull request by henry-li-06**
### Summary 

The PR includes refactoring for the API endpoint handlers for the member and team APIs that reduces the length of each endpoint handler. 
- Implemented authentication middleware so that each endpoint handler doesn't have to authenticate the user
- Implemented DAO abstraction to interact with database so that each endpoint handler does not directly call the db

### Test Plan 

N/A

### Breaking Changes  

None

### Note 
The commits/files changed look somewhat strange because there were some issues with the author of some of the earlier commits which required forcefully fixing it.

### CODE REVIEW

**Comment by SamChou19815** [comment]
Do we still plan to merge it? @henry-li-06

**Comment by henry-li-06** [comment]
> Do we still plan to merge it? @henry-li-06

I think it'd be good to get it merged. There's nothing new in this PR, just some refactoring for endpoint handlers.

File: nova-server/DataTypes.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { firestore } from 'firebase-admin';
```
Comment (SamChou19815) [inline nova-server/DataTypes.d.ts:1]: It seems to include changes from #14. Similar to #14, I'm not sure why we need this.

Comment (henry-li-06) [reply nova-server/DataTypes.d.ts]: My mistake, most likely branched off the wrong branch. I've removed the `nova-server` directory from this branch

**Review by SamChou19815 — APPROVED** [review]

**Comment by SamChou19815** [comment]
@henry-li-06 ready to merge?

**Pull request merged**